### PR TITLE
Elasticsearch Discovery with Kubernetes

### PR DESCRIPTION
**Pull request by satnam6502**
See the README.md file
This PR lets us easily ramp up and down the number of Elasticsearch nodes in a cluster using a replication controller. How cool is that?

### CODE REVIEW

**Comment by satnam6502** [comment]
People who may be interested: @lavalamp @thockin

File: examples/elasticsearch/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.PHONY: elasticsearch_discovery build push all
+
+TAG = 1.0
+
+build:
+	docker build -t satnam6502/elasticsearch:$(TAG) .
+
+push:
+	docker push satnam6502/elasticsearch:$(TAG)
```
Comment (lavalamp) [inline examples/elasticsearch/Makefile]: hm, maybe switch to kubernetes/?

File: examples/elasticsearch/README.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  user: "YWRtaW4K"
+  password: "SkU0bEtzZDdqUEZSNXBTWAo="
+
+```
+which can be used to create the secret in your namespace:
+```
+kubectl create -f apiserver-secret.yaml --namespace=mytunes
+
+```
+No you are ready to create the replication controller which will then create the pods:
```
Comment (lavalamp) [inline examples/elasticsearch/README.md]: s/No/Now/

**Comment by lavalamp** [comment]
Please take the examples/elasticsearch/elasticsearch_discovery binary out of the commit. :)

File: examples/elasticsearch/elasticsearch_discovery.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+)
+
+var (
+	username   = flag.String("username", "admin", "Username for basic authentication to the API server.")
+	password   = flag.String("password", "", "Password for basic authentication to the API server.")
+	server     = flag.String("server", "", "The address and port of the Kubernetes API server")
+	namespace  = flag.String("namespace", api.NamespaceDefault, "The namespace containing Elasticsearch pods")
+	labelKey   = flag.String("label_key", "name", "The label key to be used to filter Elasticsearch pods")
```
Comment (lavalamp) [inline examples/elasticsearch/elasticsearch_discovery.go]: Consider taking a selector as a string-- that way it's easy for user to specify multiple label/key pairs?

**Comment by lavalamp** [comment]
Looks pretty cool-- are you going to switch to secrets & tokens in the future?

File: examples/elasticsearch/Makefile (hunk 1 above)
Comment (satnam6502) [inline examples/elasticsearch/Makefile]: Done.

File: examples/elasticsearch/README.md (hunk 2 above)
Comment (satnam6502) [inline examples/elasticsearch/README.md]: Fixed, thanks.

File: examples/elasticsearch/elasticsearch_discovery.go (hunk 3 above)
Comment (satnam6502) [inline examples/elasticsearch/elasticsearch_discovery.go]: Done.

**Comment by lavalamp** [comment]
Shippable looks like a flake

**Pull request merged**